### PR TITLE
Use GolantCI-lint instead of Gometalinter

### PR DESCRIPTION
**Pull request by kevgo**
Gometalinter is officially deprecated in favor of [golangci-lint](https://github.com/golangci/golangci-lint), which has [many advantages](https://github.com/golangci/golangci-lint#comparison).

I am enabling many checks here since they encapsulate community best practices, it seems advisable to follow them since I don't feel strongly the other way. 

This PR uses `dep` that is introduced in #1195.

### CODE REVIEW

**Comment by charlierudolph** [comment]
Nice, I also switched cucumber-engine over to this recently

File: src/util/util.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func Indent(message string, level int) string {
 func Pluralize(count, word string) string {
 	result := count + " " + word
 	if count != "1" {
-		result = result + "s"
+		result += result
```
Comment (charlierudolph) [inline src/util/util.go]: Shouldn't this be `result += "s"`

Comment (kevgo) [reply src/util/util.go]: Yes, of course!

**Comment by kevgo** [comment]
@charlierudolph Whoops I hadn't realized that this didn't pass on CI. I updated Travis to Go 1.12 and now it works. Please take another look. 🙂

**Pull request merged**